### ISSUE TITLE
Disable setState warnings for didX hooks

### ISSUE THREAD

**Issue by glebec** [FullstackAcademy/eslint-config-fullstack#56]
```
'react/no-did-mount-set-state': 1, // Prevent usage of setState in componentDidMount
'react/no-did-update-set-state': 1, // Prevent usage of setState in componentDidUpdate
```

More annoying than helpful, especially with `async` funcs throwing false positives.

See https://github.com/yannickcr/eslint-plugin-react/issues/596

**Comment by fterdal** [comment]
I heartily agree with this, and I've had several students complain about this in particular.
